changelog.d/12853.docker

```diff
@@ -0,0 +1 @@
+Fix the docker file after a dependency update.
```

Comment:
This changelog entry is not very descriptive and doesn't explain why we made the change (what was broken before).

For reference, these changes are spawning from some discussion in [#synapse-dev:matrix.org](https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$3yGRxuxblgIXMXapjUE4cTDEXTVqjduYTK7eDQc2Ufc?via=matrix.org&via=element.io&via=vector.modular.im) which discusses the `ModuleNotFoundError: No module named 'matrix_common'` errors popping up when [Complement is running in CI](https://github.com/matrix-org/synapse/runs/6566243070?check_suite_focus=true#step:6:1866).

As far as I can tell, the dependency was missing in CI because of this reason mentioned in the room discussion:

> hmm, the lockfile is changing all deps to from main to dev category :/
> Yeah, that seems to be the problem, as export ignores dev deps by default :/
>
> *-- [@erikjohnston](https://matrix.to/#/!XaqDhxuTIlvldquJaV:matrix.org/$56fcBKskPHkq5hCP4xt7MXAuC4JMRBJ5rEejdqjzT0M?via=matrix.org&via=element.io&via=vector.modular.im)*

As the issue description already mentions, the root cause might be because we didn't pin `poetry-core` before and it was pulling in a recent beta release which introduced the unexpected problem behavior.

---

This information should also be in the PR description. I'm not confident in my conclusion here to update it myself.

> better PR subject and description please. The current ones will make zero sense in a few months time 😇
>
> *-- @richvdh, https://github.com/matrix-org/synapse/pull/12853#issuecomment-1135660687*